yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -21,6 +21,7 @@ import java.util.Collections
 import java.util.concurrent._
 import java.util.regex.Pattern
 
+import scala.collection.immutable.Set
```

Comment:
this is very subjective, but I find it really confusing to have fully imported both immutable and mutable collections in the same file.  (a) If you really want the immutable one, I'd just fully qualify it where you need to `collection.immutable.Set`.  Or maybe (b) you could just use a mutable Set here for consistency?  Or (c) if the code here doesn't care whether its mutable or immutable, you could take a `collection.Set`, and just fully qualify it where you use it?